docs/adr/0001-error-handling-strategy.md

```diff
@@ -0,0 +1,28 @@
+# Error handling strategy
+
+## Status
+
+Established. (The practices were in place by the time we started adding ADRs.)
+
+## Context
+
+NMEA messages containing AIS data are sometimes malformed. The messages are often transferred from transponder devices to computer equipment over a serial port, which have limited error detection and correction capabilities, making it possible for messages to lose characters or contain garbage characters.
+
+## Decision
+
+We deal with malformed NMEA messages at two levels.
+
+The `NmeaLineParser` detects failures to conform to the basic structure required of an NMEA message. It reports these errors by throwing an `ArgumentException` with a text message describing the way in which the message structure does not match the specification. For example, if the normal `!` start character is missing, the error message is `Invalid data. Expected '!' at sentence start`. In cases where the message is well formed but with features that Ais.Net cannot currently parse, we throw a `NotSupportedException` exception.
```

Comment:
I assume the `NotSupportedException` also contains a useful text message?

Reply:
Yes.